services/extension/service.go

```diff
@@ -316,7 +316,7 @@ func (s *service) deleteConfigsForParamsChange(ctx context.Context, params api.P
 }
 
 func (s *service) deleteSecretsForParamsChange(ctx context.Context, params api.Params, name, namespace string) {
-	if len(params.Secrets.Keys) == 0 {
+	if len(params.Secrets.Keys) == 0 && len(params.Container.SecretEnvironmentVariables) == 0 {
```

Comment:
do we need to check for secret envvars in sidecar here?

Reply:
yes, let me use the params.HasSecrets() there as well.